graphql-dgs/src/main/java/com/netflix/graphql/dgs/DgsMutation.java

```diff
@@ -16,12 +16,15 @@
 
 package com.netflix.graphql.dgs;
 
+import org.springframework.core.annotation.AliasFor;
+
 import java.lang.annotation.*;
 
 @Target(ElementType.METHOD)
 @Retention(RetentionPolicy.RUNTIME)
 @DgsData(parentType = "Mutation")
 @Inherited
 public @interface DgsMutation {
+    @AliasFor(annotation = DgsData.class)
```

Comment:
Spring Boot 3.0 was logging a semi-noisy warning about an upcoming deprecation for all `@DgsData` children. Example output:

> o.s.c.annotation.AnnotationTypeMapping : Support for convention-based annotation attribute overrides is deprecated and will be removed in Spring Framework 6.1. Please annotate the following attributes in @com.netflix.graphql.dgs.DgsMutation with appropriate @AliasFor declarations: [field]

The change is discussed here: https://github.com/spring-projects/spring-framework/issues/28760

> Since explicit overrides are favorable to implicit overrides, and since the support for convention-based overrides increases the complexity of Spring's annotation search algorithms, we will deprecate convention-based overrides in 6.0 and remove the support in 6.1.

For this PR, I kept `@Inherited` in place on all annotations. I did some digging into why `@Inherited` was being used and found: https://github.com/Netflix/dgs-framework/issues/1110.

In another PR, I'd be happy to implement Spring's `AnnotationUtils` to allow us to remove `@Inherited`. Just LMK!